addons/point_of_sale/static/src/js/models.js

```diff
@@ -425,11 +425,22 @@ exports.PosModel = Backbone.Model.extend({
         loaded: function(self, categories){
             self.db.add_categories(categories);
         },
+    },{
+        model: 'product.packaging',
+        fields: ['name', 'barcode', 'product_id'],
+        domain: function(self){return [['barcode', '!=', '']]; },
+        loaded: function(self, product_packagings) {
+            self.product_packagings = {};
+            _.map(product_packagings, function (product_packaging) {
+                self.product_packagings[product_packaging.id] = product_packaging;
+            });
+        }
     },{
         model:  'product.product',
         fields: ['display_name', 'lst_price', 'standard_price', 'categ_id', 'pos_categ_id', 'taxes_id',
                  'barcode', 'default_code', 'to_weight', 'uom_id', 'description_sale', 'description',
-                 'product_tmpl_id','tracking', 'write_date', 'available_in_pos', 'attribute_line_ids'],
+                 'product_tmpl_id','tracking', 'write_date', 'available_in_pos', 'attribute_line_ids',
+                 'packaging_ids'],
```

Comment:
packaging_ids not needed, because you'll grab the product.packaging with the product_id, and qty, ... And so you'll find the product through its id